App/Talent.App.WebApp/wwwroot/js/react/ReactScripts/Profile/Location.jsx

```diff
@@ -196,12 +200,41 @@ export class Address extends React.Component {
 export class Nationality extends React.Component {
     constructor(props) {
         super(props)
-       
+
+        this.handleChange = this.handleChange.bind(this);
+    }
+
+    handleChange(event){
+        if (event.target.value) {
+            const newData = {};
+            newData[event.target.name] = event.target.value;
+            this.props.saveProfileData(newData);
+        }
+        else {
+            TalentUtil.notification.show("Invalid nationality", "error", null, null);
+        }
     }
 
     
     render() {
+        const countries = Object.keys(Countries).map(c => ({value: c, title: c}));
```

Comment:
What if accidentally, the backend returns no **Countries** or it becomes a string or boolean. This condition will then cause an error, or will crash the app.

What you can do here is to verify first if **Countries** is a valid object or array then do the mapping

Reply:
Thanks Kester, I've added a new layer of safety to make sure countries will return an empty array if Countries is invalid.